src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -0,0 +1,1329 @@
+            comp.VerifyDiagnostics(
+                // (13,12): error CS8652: The feature 'parameterless struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public S1() { Y = 1; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "S1").WithArguments("parameterless struct constructors").WithLocation(13, 12),
+                // (13,12): error CS0171: Field 'S1.X' must be fully assigned before control is returned to the caller
+                //     public S1() { Y = 1; }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S1").WithArguments("S1.X").WithLocation(13, 12),
+                // (20,12): error CS0171: Field 'S2.X' must be fully assigned before control is returned to the caller
+                //     public S2(object y) { Y = y; }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S2").WithArguments("S2.X").WithLocation(20, 12));
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics(
+                // (13,12): error CS0171: Field 'S1.X' must be fully assigned before control is returned to the caller
+                //     public S1() { Y = 1; }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S1").WithArguments("S1.X").WithLocation(13, 12),
+                // (20,12): error CS0171: Field 'S2.X' must be fully assigned before control is returned to the caller
+                //     public S2(object y) { Y = y; }
+                Diagnostic(ErrorCode.ERR_UnassignedThis, "S2").WithArguments("S2.X").WithLocation(20, 12));
+        }
+
+        [Fact]
+        public void FieldInitializers_01()
+        {
+            var source =
+@"#pragma warning disable 649
+using System;
+struct S1
+{
+    object X = null;
+    object Y;
+    public override string ToString() => string.Format($""({X}, {Y})"");
+}
+struct S2
+{
+    object X = null;
+    object Y;
+    public S2() { Y = 1; }
+    public override string ToString() => string.Format($""({X}, {Y})"");
+}
+struct S3
+{
+    object X;
+    object Y = null;
+    public S3(object x) { X = x; }
+    public override string ToString() => string.Format($""({X}, {Y})"");
+}
+class Program
+{
+    static void Main()
+    {
+        Console.WriteLine(new S1());
+        Console.WriteLine(new S2());
+        Console.WriteLine(new S3());
+    }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (5,12): error CS8652: The feature 'parameterless struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     object X = null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "X").WithArguments("parameterless struct constructors").WithLocation(5, 12),
```

Comment:
nit: would it be worth splitting the feature into "parameterless struct constructors" and "struct field initializers"?